util/types/time.go

```diff
@@ -1343,17 +1343,9 @@ func checkDatetimeType(t TimeInternal, allowZeroDate bool) error {
 	return nil
 }
 
-// ExtractTimeNum extracts time value number from time unit and format.
-func ExtractTimeNum(unit string, t Time) (int64, error) {
+// ExtractDatetimeNum extracts time value number from datetime unit and format.
+func ExtractDatetimeNum(unit string, t Time) (int64, error) {
```

Comment:
how about change the order of parameter `unit` and `t`, and change `t Time` to `t *Time` ?